### DISCUSSION TITLE
Phala phat contract pallet on substrate

### DISCUSSION THREAD

**Discussion by silicology1** [paritytech/substrate#14815] [Ideas]
Provide phat contract pallet of [phala network](https://phala.network/) so that in can be used in any substrate chains for offchain computation. It would be great to also include gpu computation to host machine learning models in the substrate app chain itself.

**Comment by ggwpez** [comment]
WDYM? How are we preventing anyone from using the Phala pallet?!

**Comment by silicology1** [comment]
Ok.  😀 Thanks for reply. I will go through the code. Not sure of its pitfalls, I need it like a plugin crate, so that I don't have to maintain it.  But substrate pallets are very composable. Will work through it.
